src/util/system.cpp

```diff
@@ -591,7 +591,7 @@ bool ArgsManager::IsArgNegated(const std::string& strArg) const
 std::string ArgsManager::GetArg(const std::string& strArg, const std::string& strDefault) const
 {
     const util::SettingsValue value = GetSetting(strArg);
-    return value.isNull() ? strDefault : value.isFalse() ? "0" : value.isTrue() ? "1" : value.get_str();
+    return value.isNull() ? strDefault : value.isFalse() ? "0" : value.isTrue() ? "1" : value.getValStr();
```

Comment:
This also returns "" for an Object or Array. Is that desirable?

Reply:
> This also returns "" for an Object or Array. Is that desirable?

Good point, it does seem better not to change that behavior here, since the goal of this PR is just to prevent the GUI crashing if someone has a settings.json file with a valid int setting.

I'll update the PR to avoid changing array/object behavior here and add test coverage for it.

After this, it's probably debatable what behavior would be best. The `ArgsManager::GetSetting` method always lets you get the raw json value, but `ArgsManager::Get(Int,Bool,Path)Arg` methods are supposed to be convenient helper methods for handling default values and type coercion. If an object or array value is set and scalar value is requested, potential options seem to be:

1. Trigger an assert or exception (current behavior)
2. Return sentinel values like `""` or `0` or `false`
3. Return the default value, treat as if setting was not set
4. Do something more fancy like return the first or last value in the array, or return the object as a string.

After listing these, 1 does actually seem like the best option to me, but I wouldn't be surprised if maybe there are cases where something like 3 or 4 could make more sense.

Reply:
How about 5. fail to initialize with an error message?

Reply:
> How about 5. fail to initialize with an error message?

Good point. That really does make the most sense and would be good to implement outside of GetArg (obviously GetArg is called many places after initialization where it is too late to stop initialization). #17580 actually does most of the work of needed to implement startup rejection of arrays and objects since it adds an ALLOW_LIST flag to all list settings and disallows GetArg even being called when ALLOW_LIST is specified. #17580 currently depends on #16545, but I could probably rebase it to be independent.

To be sure though, all of this is off topic from the original issue and current PR.

This PR is just one-line change to prevent the gui aborting when a *valid* integer setting is specified because the isNum() case was unhandled. The diff that luke was responding to accidentally changed the behavior for arrays and objects as well as numbers, but the current PR strictly limits the change to the isNum case and doesn't affect arrays or objects.

Reply:
Yes, just wanted to note that exiting with an error is better than a crash.